doc/vector-extra/riscv-crypto-vector-extra-zvbc32e.adoc

```diff
@@ -0,0 +1,23 @@
+[[zvbc32e,Zvbc32e]]
+=== `Zvbc32e` - Vector Carryless Multiplication
```

Comment:
Maybe "32-bit Vector Carryless Multiplication"?

Also, are there specific use cases in mind where this feature would be useful?  https://github.com/riscv/riscv-crypto/issues/309 mentioned CRC-32, but I'm not sure why that would be true.  CRC-32 implementations that use "folding" with carryless multiplication multiply segments of the data by a 32-bit multiplicand, but those segments of data can be any length.  Other CPUs (and the currently ratified Zvbc extension) provide 64-bit carryless multiplication, so CRC-32 implementations typically multiply 64 bits of data by a 32-bit multiplier to get a 95-bit result.  You *could* multiply 32 bits of data by a 32-bit multiplier to get a 63-bit result, but that would result in twice as many multiplications being needed to compute the CRC-32, as each one would process half as much data.  That probably wouldn't be faster.

A similar observation applies to e.g. CRC-16 where the multiplications are typically 64-bit by 16-bit.

BTW, if adding 32-bit, why not also add 16-bit and 8-bit?  What is special about 32?

Reply:
Those are very good points.
There are multiple considerations at play here.
- 32-bit CLM allows vector implementation with smaller ELEN=32 to implement it (which was not possible for `Zvbc`)
- 64-bit or 32-bit CLM for folding CRC does not make a performance difference as far as I can tell but should make a power difference. It is true that in folding CRC the data segment block can be of arbitrary length but the constant multiplicand is of the size of the CRC so in the case of a CRC-32 at least half the 64-bit CLM would be unused (which might not be an issue on all micro-arch, but 32-bit CRC should allow easier determination of expected multiplier activity).

And finally there is nothing special about 32-bit. It was initially introduced because of the ELEN=32 case, but we listed in the questions above
>  Should Zvbc32e support SEW=16 ? (SEW=8 ?)

I think there is a good case for supporting other SEW:
- no need encoding required
- should allow other use cases

If we want to go that route the question is:
- Should we split different SEW support into different sub-extensions (`Zvbc16e`, `Zvbc8e`) so people can pick and chose ? I think this might not be such a good idea because I don't expect supporting smaller element width to have a big cost (although I did not measure it yet) and the added complexity of numerous extensions does not seem worth it.

(One of the option which was discarded for the fast track was suggesting a widening CLM to merge `vclmul` and `vclmulh` together, but it seems too big of a change to fit into a fast track and the potential impact of having to switch SEW back and forth was not yet properly evaluated on the target use cases).